// dao/attendanceDao.js
var mysql=require('mysql');
var moment = require('moment');

function connectServer(){
  var client=mysql.createConnection({
    host:'localhost',
         user:'root',
         password:'123',
         database:'test_node'
     })
     return client;
 }

//查询考勤记录
function  selectFun(client,name,month,type,callback){
    //client为一个mysql连接对象
    var selectSql = 'select * from attendance where name="'+name+'" and month="'+month+'"';
    if(type != -1){
      selectSql += 'and type="'+type+'"';
    }
     client.query(selectSql,function(err,results,fields){
         if(err) throw err;
         if(results && results.length){
            for(var d = 0;d < results.length;d++){//格式化日期
                results[d].date = moment(results[d].date).format("YYYY-MM-DD");
            }
         }
         var info = {
           success: true,
           data: results
         }
         callback(info);
    });
 }

//新增考勤记录
function insertFun(client,name,month,date,start_time,end_time,duration,type,callback){
     client.query('insert into attendance(name,month,date,start_time,end_time,duration,type) values(?,?,?,?,?,?,?)', [name,month,date,start_time,end_time,duration,type], function(err,result){
        if( err ){
            console.log( "error:" + err.message);
             return err;
         }
           callback(err);
     });
}

//selectOverTimeDateFun

//checkOverTimeFun


exports.connect = connectServer;
exports.selectFun  = selectFun;
exports.insertFun = insertFun;
